dosbox-x 2022.12.26 build failure

Looks like there are some DEFINE missing

```
int_dosv.cpp:1853:19: error: use of undeclared identifier 'FALSE'
                                IME_SetEnable(FALSE);
                                              ^
int_dosv.cpp:1855:19: error: use of undeclared identifier 'TRUE'
                                IME_SetEnable(TRUE);
                                              ^
int_dosv.cpp:1857:7: error: use of undeclared identifier 'IME_GetEnable'
                        if(IME_GetEnable())
                           ^
int_dosv.cpp:2297:17: error: use of undeclared identifier 'TRUE'
                IME_SetEnable(TRUE);
                              ^
int_dosv.cpp:2304:17: error: use of undeclared identifier 'FALSE'
                IME_SetEnable(FALSE);
                              ^
int_dosv.cpp:2318:7: error: use of undeclared identifier 'IME_GetEnable'
                        if(IME_GetEnable()) {
                           ^
6 errors generated.
```

relates to https://github.com/Homebrew/homebrew-core/pull/119225

--- a/src/gui/sdlmain.cpp
+++ b/src/gui/sdlmain.cpp
@@ -3062,12 +3062,12 @@ void GFX_Stop() {
     sdl.active=false;
 }
 
-#if (defined(WIN32) && !defined(HX_DOS) || defined(LINUX) && C_X11 || defined(MACOSX)) && (defined(C_SDL2) || defined(SDL_DOSBOX_X_SPECIAL))
+#if (defined(WIN32) && !defined(HX_DOS) || defined(LINUX) && C_X11 || defined(MACOSX)) && ((defined(C_SDL2) && defined(SDL_DOSBOX_X_IME)) || defined(SDL_DOSBOX_X_SPECIAL))
 static uint8_t im_x, im_y;
 #endif
 
 void GFX_Start() {
-#if defined(MACOSX) && !defined(C_SDL2)
+#if defined(MACOSX) && !defined(C_SDL2) && defined(SDL_DOSBOX_X_SPECIAL)
 	if(dos.im_enable_flag) {
 		SDL_SetIMValues(SDL_IM_ENABLE, 1, NULL);
 		im_x = -1;
@@ -5144,7 +5144,7 @@ bool sdl_wait_on_error() {
     return sdl.wait_on_error;
 }
 
-#if (defined(WIN32) && !defined(HX_DOS) || defined(LINUX) && C_X11 || defined(MACOSX)) && (defined(C_SDL2) || defined(SDL_DOSBOX_X_SPECIAL))
+#if (defined(WIN32) && !defined(HX_DOS) || defined(LINUX) && C_X11 || defined(MACOSX)) && ((defined(C_SDL2) && defined(SDL_DOSBOX_X_IME)) || defined(SDL_DOSBOX_X_SPECIAL))
 static uint32_t last_ticks;
 void SetIMPosition() {
 	uint8_t x, y;
@@ -5414,7 +5414,7 @@ void GFX_Events() {
     if (sdl.desktop.type == SCREEN_GAMELINK) OUTPUT_GAMELINK_InputEvent();
 #endif
 
-#if (defined(WIN32) && !defined(HX_DOS) || defined(LINUX) && C_X11 || defined(MACOSX)) && (defined(C_SDL2) || defined(SDL_DOSBOX_X_SPECIAL))
+#if (defined(WIN32) && !defined(HX_DOS) || defined(LINUX) && C_X11 || defined(MACOSX)) && ((defined(C_SDL2) && defined(SDL_DOSBOX_X_IME)) || defined(SDL_DOSBOX_X_SPECIAL))
    if(IS_PC98_ARCH) {
        static uint32_t poll98_delay = 0;
        uint32_t time = GetTicks();
@@ -5842,7 +5842,7 @@ void GFX_Events() {
         }
     }
 #endif
-#if (defined(WIN32) && !defined(HX_DOS) || defined(LINUX) && C_X11 || defined(MACOSX)) && (defined(C_SDL2) || defined(SDL_DOSBOX_X_SPECIAL))
+#if (defined(WIN32) && !defined(HX_DOS) || defined(LINUX) && C_X11 || defined(MACOSX)) && ((defined(C_SDL2) && defined(SDL_DOSBOX_X_IME)) || defined(SDL_DOSBOX_X_SPECIAL))
    if(IS_PC98_ARCH) {
        static uint32_t poll98_delay = 0;
        uint32_t time = GetTicks();
@@ -6177,7 +6177,7 @@ void GFX_Events() {
 			}
 #endif
 #endif
-#if defined (MACOSX)
+#if defined (MACOSX) &&  defined(SDL_DOSBOX_X_SPECIAL)
 			int onoff;
 			if(SDL_GetIMValues(SDL_IM_ONOFF, &onoff, NULL) == NULL) {
 				if(onoff != 0 && event.type == SDL_KEYDOWN) {

--- a/src/ints/bios.cpp
+++ b/src/ints/bios.cpp
@@ -5829,7 +5829,7 @@ static Bitu INT8_Handler(void) {
     mem_writed(BIOS_TIMER,value);
 
 	if(bootdrive>=0) {
-#if (defined(WIN32) && !defined(HX_DOS) || defined(LINUX) && C_X11 || defined(MACOSX)) && (defined(C_SDL2) || defined(SDL_DOSBOX_X_SPECIAL))
+#if (defined(WIN32) && !defined(HX_DOS) || defined(LINUX) && C_X11 || defined(MACOSX)) && ((defined(C_SDL2) && defined(SDL_DOSBOX_X_IME)) || defined(SDL_DOSBOX_X_SPECIAL))
         SetIMPosition();
 #endif
     } else if (IS_DOSV && DOSV_CheckCJKVideoMode()) {
@@ -5837,7 +5837,7 @@ static Bitu INT8_Handler(void) {
     } else if(J3_IsJapanese()) {
         INT8_J3();
     } else if (IS_DOS_CJK) {
-#if (defined(WIN32) && !defined(HX_DOS) || defined(LINUX) && C_X11 || defined(MACOSX)) && (defined(C_SDL2) || defined(SDL_DOSBOX_X_SPECIAL))
+#if (defined(WIN32) && !defined(HX_DOS) || defined(LINUX) && C_X11 || defined(MACOSX)) && ((defined(C_SDL2) && defined(SDL_DOSBOX_X_IME)) || defined(SDL_DOSBOX_X_SPECIAL))
         SetIMPosition();
 #endif
     }

--- a/src/ints/bios_keyboard.cpp
+++ b/src/ints/bios_keyboard.cpp
@@ -1326,7 +1326,7 @@ Bitu INT16_Handler(void) {
                 }
             }
         }
-#elif (defined(WIN32) && !defined(HX_DOS) || defined(MACOSX)) && defined(C_SDL2)
+#elif (defined(WIN32) && !defined(HX_DOS) || defined(MACOSX)) && defined(C_SDL2) && defined(SDL_DOSBOX_X_IME)
 #if defined(USE_TTF)
         if((IS_DOSV || ttf_dosv) && IS_DOS_CJK && (DOSV_GetFepCtrl() & DOSV_FEP_CTRL_IAS)) {
 #else

--- a/src/ints/int_dosv.cpp
+++ b/src/ints/int_dosv.cpp
@@ -1847,7 +1847,7 @@ static Bitu mskanji_api(void)
 					real_writew(param_seg, param_off + 2, 0x0009);
 			}
 		}
-#elif (defined(WIN32) && !defined(HX_DOS) || defined(MACOSX)) && defined(C_SDL2)
+#elif (defined(WIN32) && !defined(HX_DOS) || defined(MACOSX)) && defined(C_SDL2) && defined(SDL_DOSBOX_X_IME)
 		if(mode & 0x8000) {
 			if(mode & 0x0001)
 				IME_SetEnable(FALSE);
@@ -2293,14 +2293,14 @@ Bitu INT6F_Handler(void)
 	case 0x05:
 #if (defined(WIN32) && !defined(HX_DOS) || defined(LINUX) && C_X11 || defined(MACOSX)) && !defined(C_SDL2) && defined(SDL_DOSBOX_X_SPECIAL)
 		SDL_SetIMValues(SDL_IM_ONOFF, 1, NULL);
-#elif (defined(WIN32) && !defined(HX_DOS) || defined(MACOSX)) && defined(C_SDL2)
+#elif (defined(WIN32) && !defined(HX_DOS) || defined(MACOSX)) && defined(C_SDL2) && defined(SDL_DOSBOX_X_IME)
 		IME_SetEnable(TRUE);
 #endif
 		break;
 	case 0x0b:
 #if (defined(WIN32) && !defined(HX_DOS) || defined(LINUX) && C_X11 || defined(MACOSX)) && !defined(C_SDL2) && defined(SDL_DOSBOX_X_SPECIAL)
 		SDL_SetIMValues(SDL_IM_ONOFF, 0, NULL);
-#elif (defined(WIN32) && !defined(HX_DOS) || defined(MACOSX)) && defined(C_SDL2)
+#elif (defined(WIN32) && !defined(HX_DOS) || defined(MACOSX)) && defined(C_SDL2) && defined(SDL_DOSBOX_X_IME)
 		IME_SetEnable(FALSE);
 #endif
 		break;
@@ -2314,7 +2314,7 @@ Bitu INT6F_Handler(void)
 					reg_al = 0x01;
 				}
 			}
-#elif (defined(WIN32) && !defined(HX_DOS) || defined(MACOSX)) && defined(C_SDL2)
+#elif (defined(WIN32) && !defined(HX_DOS) || defined(MACOSX)) && defined(C_SDL2) && defined(SDL_DOSBOX_X_IME)
 			if(IME_GetEnable()) {
 				reg_al = 0x01;
 			}

--- a/src/shell/shell.cpp
+++ b/src/shell/shell.cpp
@@ -992,7 +992,7 @@ void DOS_Shell::Prepare(void) {
         initcodepagefont();
         dos.loaded_codepage=cp;
     }
-#if (defined(WIN32) && !defined(HX_DOS) || defined(LINUX) && C_X11 || defined(MACOSX)) && (defined(C_SDL2) || defined(SDL_DOSBOX_X_SPECIAL))
+#if (defined(WIN32) && !defined(HX_DOS) || defined(LINUX) && C_X11 || defined(MACOSX)) && ((defined(C_SDL2) && defined(SDL_DOSBOX_X_IME)) || defined(SDL_DOSBOX_X_SPECIAL))
     if (enableime) SetIMPosition();
 #endif
 }

